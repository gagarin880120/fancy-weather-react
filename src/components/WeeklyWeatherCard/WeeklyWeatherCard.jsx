import React from 'react';
import PropTypes from 'prop-types';
import * as Icons from '@intern0t/react-weather-icons';
import styles from './WeeklyWeatherCard.module.css';

export default function WeeklyWeatherCard({ weekday, temp, icon }) {
  const Icon = Icons[icon];
  return (
    <div className={styles.wrapper}>
      <span className={styles.weekday}>
        {weekday}
      </span>
      <div className={styles.temperature}>
        {temp}
      </div>
      <Icon color="#000" size={64} />
    </div>
  );
}

WeeklyWeatherCard.propTypes = {
  weekday: PropTypes.string,
  temp: PropTypes.number,
  icon: PropTypes.string,
};

WeeklyWeatherCard.defaultProps = {
  weekday: '',
  temp: 0,
  icon: '',
};
