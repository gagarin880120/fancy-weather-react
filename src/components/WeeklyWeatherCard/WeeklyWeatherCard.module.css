.wrapper {
  margin-right: 10px;
}

.weekday {
  text-transform: capitalize;
  font-size: 20px;
}

.temperature {
  margin-top: 5px;
  font-size: 54px;
}
