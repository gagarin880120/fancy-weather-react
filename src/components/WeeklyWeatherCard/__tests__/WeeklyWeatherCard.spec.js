import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import WeeklyWeatherCard from '../WeeklyWeatherCard';

jest.mock('@intern0t/react-weather-icons', () => ({sunny: jest.fn()}));

describe('WeeklyWeatherCard', () => {
  describe('component', () => {
    test('should render without crashing', () => {
      const renderer = new ShallowRenderer();
      renderer.render(
        <WeeklyWeatherCard
          weekday="Tuesday"
          temp={20}
          icon="sunny"
        />
      );

      const result = renderer.getRenderOutput();

      expect(result).toMatchSnapshot();
    });
  });
});
