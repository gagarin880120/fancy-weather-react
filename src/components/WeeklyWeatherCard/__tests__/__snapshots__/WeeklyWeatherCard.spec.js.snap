// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeeklyWeatherCard component should render without crashing 1`] = `
<div
  className="wrapper"
>
  <span
    className="weekday"
  >
    Tuesday
  </span>
  <div
    className="temperature"
  >
    20
    <mockConstructor
      color="#fff"
      size={64}
    />
  </div>
</div>
`;
